controllers/upload.go

```diff
@@ -0,0 +1,47 @@
+package controllers
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/gin-gonic/gin"
+	"net/http"
+	"zestream-server/constants"
+	"zestream-server/types"
+	"zestream-server/utils"
+)
+
+func GeneratePresignedURL(c *gin.Context) {
+
+	// Validate the incoming request
+	var request types.GenerateURLRequest
+	if err := c.ShouldBindJSON(&request); err != nil {
```

Comment:
the GET request wont have json, it would be better to fetch from query param and extract extension from fileName and add it to video ID

Reply:
Done via https://github.com/ZeStream/zestream-server/pull/4/commits/5d8762e1eb3a7de189cad0f11c6d4fd2bba8fd09


<img width="1193" alt="Screenshot 2022-12-31 at 12 16 25 AM" src="https://user-images.githubusercontent.com/72073401/210102918-d47a3d38-9516-4ac7-97e8-b7a0c82f9d6f.png">

----
<img width="1190" alt="Screenshot 2022-12-31 at 12 17 58 AM" src="https://user-images.githubusercontent.com/72073401/210103032-d0eb0288-6080-41ae-bbf6-d1384793f4b1.png">